### PR TITLE
Portfolio and coin page

### PR DESCRIPTION
**Pull request by Basiol97**
The portfolio page, coin page and nav-bar component were built and styled. In addition, major styling (responsiveness) improvements were made to the CoinList page.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,55 +1,124 @@
 import React from "react";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import { useState, useEffect } from "react";
 import { createGlobalStyle, ThemeProvider } from "styled-components";
 import { CoinList, CoinPage, Portfolio } from "pages";
-import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
-
+import NavBar from "components/NavBar";
```
Comment (yoloOnTheBattlefield) [inline src/App.js:6]: new line

File: src/App.js [hunk 2]
```diff
@@ -1,55 +1,124 @@
...
+    <ThemeProvider theme={themeMode ? lightTheme : darkTheme}>
+      <div className="App">
+        <GlobalStyle />
+        <Router>
+          <NavBar
+            handleMobileSearchClick={handleMobileSearchClick}
+            showSearchBar={showSearchBar}
+            currencySymbol={currencySymbol}
+            handleCurrencyChange={handleCurrencyChange}
+            ConverMode={ConverMode}
```
Comment (yoloOnTheBattlefield) [inline src/App.js:86]: camelCase

File: src/components/Description/index.js [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
+      <DescriptionInnerContainer>
+        <DescriptionIcon>
+          <BiLayer size={30} />
+        </DescriptionIcon>
+        <CoinDescription>
+          {text?.length && (
+            <ReadMoreReact
+              readMoreText={"Click here to read more..."}
+              text={text}
+            />
```
Comment (yoloOnTheBattlefield) [inline src/components/Description/index.js:26]: dangerously set inner html

File: src/components/MobileNavbar/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <MobileNavbarOuterContainer>
+      <MobileNavbarInnerContainer>
+        <Overview to="/">
+          {" "}
+          <FiHome size={20} /> Overview
+        </Overview>
+        <Portfolio to="portfolio">
+          <AiOutlineOrderedList size={20} /> Portfolio{" "}
+        </Portfolio>
+        <Search onClick={() => props.handleMobileSearchClick()}>
```
Comment (yoloOnTheBattlefield) [inline src/components/MobileNavbar/index.js:24]: <Search onClick={ props.handleMobileSearchClick}>

File: src/components/MobileNavbar/styled.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+export const MobileNavbarOuterContainer = styled.div`
+  display: none;
+  position: fixed;
+  z-index: 100;
+  bottom: 0;
+  width: 100%;
+  background-color: ${({ theme }) => theme.main};
+  @media(max-width: ${({ theme }) => theme.mobile}){
+display: block;
+}
```
Comment (yoloOnTheBattlefield) [inline src/components/MobileNavbar/styled.js:13]: indented

File: src/components/MobileNavbar/styled.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ `;
+
+export const MobileNavbarInnerContainer = styled.div`
+display: flex;
+justify-content: space-evenly;
+margin: 0 auto;
+
+-webkit-box-shadow: 0px 0px 10px 3px rgba(0,0,0,0.3);
+-moz-box-shadow: 0px 0px 10px 3px rgba(0,0,0,0.3);
+box-shadow: 0px 0px 10px 3px rgba(0,0,0,0.3);
```
Comment (yoloOnTheBattlefield) [inline src/components/MobileNavbar/styled.js:23]: same here

File: src/components/NavBar/index.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+                    onChange={(e) => handleChange(e)}
+                    value={serachInput}
+                    autoComplete="off"
+                  />
+                </SearchFieldOuterContainer>
+                {serachInput && (
+                  <Options onClick={() => props.handleMobileSearchClick()}>
+                    {dropDown
+                      .filter((item) => {
+                        if (item == "") {
```
Comment (yoloOnTheBattlefield) [inline src/components/NavBar/index.js:136]: double check this

File: src/components/NavBar/index.js [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+                          return item;
+                        }
+                      })
+                      .map((item) => (
+                        <div key={item.id}>
+                          <OptionLink
+                            onClick={() => {
+                              setSerachInput("");
+                            }}
+                            to={`coinpage/${item.id}`}
```
Comment (yoloOnTheBattlefield) [inline src/components/NavBar/index.js:148]: `/coinpage/${item.id}`

File: src/components/NavBar/index.js [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+                  <AiFillDollarCircle color="#06D554" size={40} />
+                ) : props.currencySymbol == "£" ? (
+                  <AiOutlinePoundCircle color="#06D554" size={40} />
+                ) : props.currencySymbol == "€" ? (
+                  <AiOutlineEuroCircle color="#06D554" size={40} />
+                ) : (
+                  <AiFillDollarCircle color="#06D554" size={40} />
+                )}
+                <CurrencyDropDownField
+                  onChange={(e) => props.handleCurrencyChange(e)}
```
Comment (yoloOnTheBattlefield) [inline src/components/NavBar/index.js:168]: onChange={props.handleCurrencyChange}

File: src/components/NavBar/index.js [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+                >
+                  {currencyOptions.map((item, index) => (
+                    <option key={index}>{item}</option>
+                  ))}
+                </CurrencyDropDownField>
+              </CurrancyContainer>
+              <ThemeContainer onClick={() => props.ConverMode()}>
```
Comment (yoloOnTheBattlefield) [inline src/components/NavBar/index.js:175]: make it shorter

File: src/pages/CoinList/CoinList.js [hunk 11]
```diff
@@ -1,76 +1,121 @@
...
+  const [defaultPeriod, setdefaultPeriod] = useState(180);
 
-class CoinList extends React.Component {
+  const fetchData = async () => {
+    setisLoading(true);
+    try {
+      const { data: response } = await axios.get(
+        `https://api.coingecko.com/api/v3/coins/bitcoin/market_chart?vs_currency=${props.currencyType}&days=180&interval=daily`
+      );
+      setfetedData(response);
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/CoinList.js:32]: typo

File: src/pages/CoinPage/CoinPage.js [hunk 12]
```diff
@@ -1,89 +1,106 @@
...
-    this.setState({ isLoading: true });
-    this.fetchMoreCoins();
-  }
+  const fetchData = async () => {
+    setisLoading(true);
+    try {
+      const { data: response } = await axios.get(
+        `https://api.coingecko.com/api/v3/coins/${coinid}?localization=false&tickers=false&market_data=true&community_data=true&developer_data=false&sparkline=false`
+      );
+      setfetedData(response);
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage/CoinPage.js:31]: typo

File: src/pages/Portfolio/Portfolio.js [hunk 13]
```diff
@@ -1,9 +1,191 @@
...
+    setisLoading(false);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    fetchData();
+  }, [props.currencyType]);
```
Comment (yoloOnTheBattlefield) [inline src/pages/Portfolio/Portfolio.js:58]: useEffect(fetchData, [props.currencyType]);

Comment (yoloOnTheBattlefield) [reply src/pages/Portfolio/Portfolio.js:58]: try this

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Basiol97) [reply src/App.js:6]: Resloved

Comment (Basiol97) [reply src/App.js:6]: Resolved

File: src/App.js (hunk 2 above)
Comment (Basiol97) [reply src/App.js:86]: Resolved

File: src/components/MobileNavbar/index.js (hunk 4 above)
Comment (Basiol97) [reply src/components/MobileNavbar/index.js:24]: Resolved

File: src/components/MobileNavbar/styled.js (hunk 5 above)
Comment (Basiol97) [reply src/components/MobileNavbar/styled.js:13]: Resolved

File: src/components/NavBar/index.js (hunk 9 above)
Comment (Basiol97) [reply src/components/NavBar/index.js:168]: Resolved

File: src/pages/CoinList/CoinList.js (hunk 11 above)
Comment (Basiol97) [reply src/pages/CoinList/CoinList.js:32]: Resolved

File: src/pages/CoinPage/CoinPage.js (hunk 12 above)
Comment (Basiol97) [reply src/pages/CoinPage/CoinPage.js:31]: Resolved

File: src/pages/Portfolio/Portfolio.js (hunk 13 above)
Comment (Basiol97) [reply src/pages/Portfolio/Portfolio.js:58]: Solution didn't work as expected

File: src/components/MobileNavbar/styled.js (hunk 6 above)
Comment (Basiol97) [reply src/components/MobileNavbar/styled.js:23]: Resolved

File: src/components/NavBar/index.js (hunk 7 above)
Comment (Basiol97) [reply src/components/NavBar/index.js:136]: Resolved

File: src/components/NavBar/index.js (hunk 8 above)
Comment (Basiol97) [reply src/components/NavBar/index.js:148]: Resolved

**Pull request merged**